.swiftlint.yml

```diff
@@ -1,86 +1,2 @@
-excluded:
-  - Pods
-  - Scripts
-  - vendor
-  - DerivedData
-
-# Rules
-only_rules:
-  # Colons should be next to the identifier when specifying a type.
-  - colon
-
-  # There should be no space before and one after any comma.
-  - comma
-
-  # if,for,while,do statements shouldn't wrap their conditionals in parentheses.
-  - control_statement
-
-  # Arguments can be omitted when matching enums with associated types if they
-  # are not used.
-  - empty_enum_arguments
-
-  # Prefer `() -> ` over `Void -> `.
-  - empty_parameters
-
-  # MARK comment should be in valid format.
-  - mark
-
-  # Opening braces should be preceded by a single space and on the same line as
-  # the declaration.
-  - opening_brace
-
-  # Files should have a single trailing newline.
-  - trailing_newline
-
-  # Lines should not have trailing semicolons.
-  - trailing_semicolon
-
-  # Lines should not have trailing whitespace.
-  - trailing_whitespace
-
-  - custom_rules
-
-# Rules configuration
-
-control_statement:
-  severity: error
-
-custom_rules:
-
-  natural_content_alignment:
-    name: "Natural Content Alignment"
-    regex: '\.contentHorizontalAlignment(\s*)=(\s*)(\.left|\.right)'
-    message: "Forcing content alignment left or right can affect the Right-to-Left layout. Use naturalContentHorizontalAlignment instead."
-    severity: warning
-
-  natural_text_alignment:
-    name: "Natural Text Alignment"
-    regex: '\.textAlignment(\s*)=(\s*).left'
-    message: "Forcing text alignment to left can affect the Right-to-Left layout. Consider setting it to `natural`"
-    severity: warning
-
-  inverse_text_alignment:
-    name: "Inverse Text Alignment"
-    regex: '\.textAlignment(\s*)=(\s*).right'
-    message: "When forcing text alignment to the right, be sure to handle the Right-to-Left layout case properly, and then silence this warning with this line `// swiftlint:disable:next inverse_text_alignment`"
-    severity: warning
-
-  localization_comment:
-    name: "Localization Comment"
-    regex: 'NSLocalizedString([^,]+,\s+comment:\s*"")'
-    message: "Localized strings should include a description giving context for how the string is used."
-    severity: warning
-
-  string_interpolation_in_localized_string:
-    name: "String Interpolation in Localized String"
-    regex: 'NSLocalizedString\("[^"]*\\\(\S*\)'
-    message: "Localized strings must not use interpolated variables. Instead, use `String(format:`"
-    severity: error
-
-  swiftui_localization:
-    name: "SwiftUI Localization"
-    regex: 'LocalizedStringKey'
-    message: "Using `LocalizedStringKey` is incompatible with our tooling and doesn't allow you to provide a hint/context comment for translators either. Please use `NSLocalizedString` instead, even with SwiftUI code."
-    severity: error
-    excluded: '.*Widgets/.*'
+parent_config: https://raw.githubusercontent.com/Automattic/swiftlint-config/0f8ab6388bd8d15a04391825ab125f80cfb90704/.swiftlint.yml
+remote_timeout: 10.0
```

Comment:
If you check the raw file, you'll see the rules are the same.

The URL contains the commit SHA1, which allows us to pin the configuration version and adopt new rules introduced in the config at our own pace here.

Reply:
I'm wondering if the `excluded` part should be left in the project, so that the swiftlint.yml file only contains rules and each project gets to decide what files the rules should be applied to.

Reply:
The configurations are hierarchical, so each project can overlay its own settings if needed. 

Admittedly, I tried the override for only rules when creating the project, not the files. However the [SwiftLint docs](https://github.com/realm/SwiftLint/tree/b9e5cfb2027bb94a0bdcafe5d757827a236d0f1b#child--parent-configs-locally) specifically say that "When merging parent and child configs, included and excluded configurations are processed carefully to account for differences in the directory location of the containing configuration files," and that makes me confident.

Reply:
> Do you think that with this capability we could be fine with shipping a default set of `excluded` folders in the shared config?

On thing we might want to reconsider, though, is _what_ to have in the `excluded` list. The [current](https://github.com/Automattic/swiftlint-config/blob/0f8ab6388bd8d15a04391825ab125f80cfb90704/.swiftlint.yml#L3-L10) values are:

```yml
  - DerivedData
  - fastlane
  - Pods
  - Scripts
  # Automattic's CI caching setup may generate this in the project folder
  - Users/builder/Library/Caches/CocoaPods/Pods
  - vendor
```

I find them all appropriate default choices apart maybe for `Scripts/`. If a project makes heavy use of local Swift scripts, it ought to lint them as well.

Besides, unlike all the other paths which are quite standard, "Scripts" is a bit arbitrary. I'm not sure whether all our apps use that name for the folder that contains tooling scripts 🤔

Reply:
My think is, only the rules and their configurations are worth sharing among projects, in terms of unifying the coding style among projects. The `excluded` part in the shared configuration feels like not that related to coding style. But anyway, I don't really have a strong opinion on this. The current configuration looks good to me 👍 

BTW, is the path `Users/builder/Library/Caches/CocoaPods/Pods` supposed to be an absolute path? And, why it needs to be in the `excluded` list? Does swiftlint somehow look for files in this cache directory?

Reply:
> My think is, only the rules and their configurations are worth sharing among projects, in terms of unifying the coding style among projects.

I can see your point there. 🤔 

I guess where I came from was more from a DRY point of view, but I wonder whether that might be pushing it too far now, given your perspective.

I'd like to merge as is and then iterate on whether to keep all or some of the `excluded` in the shared repo.

Reply:
> BTW, is the path `Users/builder/Library/Caches/CocoaPods/Pods` supposed to be an absolute path? And, why it needs to be in the `excluded` list? Does swiftlint somehow look for files in this cache directory?

The path is intentionally local. That is `./Users/` (shorted to `Users/`) instead of `/Users/`.

I didn't add more details in the commit message, but I found the PR that caused me to add that setting: https://github.com/Automattic/pocket-casts-ios/pull/349#issuecomment-1267815239

Reply:
Thanks for digging the commit up. 🙇 The relative path is the same as the the CocoaPods cache directory, but in the git checkout directory, which _feels_ like a harmless bug to me...

Reply:
Moved to https://github.com/Automattic/swiftlint-config/issues/1